doc/rst/language/spec/statements.rst

```diff
@@ -927,6 +929,208 @@ only applicable to modules, please see :ref:`Using_Modules`.
+          writeln("In M1's foo.");
+        }
+      }
+
+      module M2 {
+        proc main() {
+          import M1.{foo};
+
+          writeln("In M2's main.");
+          foo();
+        }
+      }
+
+   which also prints out
+
+   .. code-block:: printoutput
+
+      In M2's main.
+      In M1's foo.
+
+The ``module-or-symbol-rename`` or ``module-or-symbol-base`` in an ``import``
+statement must begin with one of the following:
+
+ * a top-level module name
+ * a module name currently in scope due to another ``use`` or ``import``
+   statement
+ * any number of ``super`` components to indicate a number of parents of the
+   current module (e.g. ``super.super.SomeModule``)
+ * ``this`` to indicate the requested module is a submodule of the current
+   module
+
+A submodule may not be imported without either the full path to it, or a
+``super`` or ``this`` prefix at the beginning of the path.
+
+The names that are imported by an import statement are inserted in to a new
+scope that immediately encloses the scope within which the statement appears.
+This implies that the position of the import statement within a scope has no
+effect on its behavior.  If a scope includes multiple import statements, or a
+combination of import and use statements, then the imported names are inserted
+into a common enclosing scope.
+
+An error is signalled if multiple public module-level symbols would be inserted
+into this enclosing scope with the same name, and that name is referenced by
+other statements in the same scope as the import.
+
+A module or a public module-level symbol being imported may optionally be given
+a new name using the ``as`` keyword.  This new name will be usable from the
+scope of the import in place of the old name.  This new name does not affect
+imports or uses of that module from other contexts.
+
+Import statements may be explicitly declared ``public`` or ``private``.  By
+default, imports are ``private``.  Making an import ``public`` causes its
+symbols to be visible as though they were defined in the scope with the import,
+a strategy which will be referred to as re-exporting.  However, symbols with the
+same name in the scope with the import will still take precedence.  If module A
+imports module B, and module B contains a public import of module C, then C will
+be visible to A as though it was a submodule of B.  This means that A could
+contain references like ``B.C.cSymbol`` if cSymbol was a symbol defined in C,
+regardless of if C was actually a submodule of B.  Similarly, if module B
+contains a public import of some public symbols defined in module C, then those
+symbols will be visible to A as though they were defined in module B, unless
+they are shadowed by symbols of the same name in B.  This means that A could
+contain references like ``B.cSymbol`` and it would reference C's cSymbol.
+Conversely, if B's import of C is ``private`` then A will not be able to see C's
+symbols due to that ``import``.
```

Comment:
if you can throw a quick example here, it may be nice

Reply:
I'm tempted to let the later updates to the modules primer handle it instead of inserting it here and there, would that be okay?

Reply:
sounds good